services/swagger.js

```diff
@@ -134,31 +135,37 @@ exports.addFormat = function (format, validationFunction) {
     tv4.addFormat(format, validationFunction);
 };
 
-function overrideInheritedProperties(definitions) {
+function flattenModelDefinitions(definitions) {
     Object.keys(definitions).forEach(function (defn) {
-        var props = {};
+        var flattenedDefn = {};
         if (definitions[defn].hasOwnProperty('allOf')) {
             for (var i = 0; i < definitions[defn].allOf.length; i++) {
                 //have to clone so that inherited definitions are not affected
-                definitions[defn].allOf[i] = swaggerUtil.deepClone(definitions[defn].allOf[i]);
-                overrideProps(definitions[defn].allOf[i], props);
+                definitions[defn].allOf[i] = _.cloneDeep(definitions[defn].allOf[i]);
+                flattenModel(definitions[defn].allOf[i], flattenedDefn);
             }
-            definitions[defn].properties = props;
+            flattenedDefn.required = Object.keys(flattenedDefn.required);
+            definitions[defn] = flattenedDefn;
+            delete definitions[defn].allOf;
         }
     });
 }
 
-function overrideProps(item, props) {
+function flattenModel(item, props) {
+    //have to convert 'required' array to object for merging purposes
+    if (item.required && Array.isArray(item.required)) {
+        var requiredObj = {};
+        item.required.reduce(function (prevVal, currentVal, idx) {
```

Comment:
I don't think `requiredObj` is required, also ... let's use more meaningful names.
The body of the `if` statement should look something like:
```javascript
definitionLayer.required = definitionLayer.required.reduce(function (reqsAsObj, reqdPropName, arrayIndex) {
    reqsAsObj[reqdPropName] = arrayIndex;
    return reqsAsObj;
  }, {});
```